### PR TITLE
Reduce three dependencies

### PR DESCRIPTION
**Pull request by cthoyt**
This PR reduces three dependencies:

1. Removes the backport of `importlib.metadata`. Since SSSOM-py is now py38+, this is part of the standard library
2. Hides the import for networkx inside the only function that uses it
3. Hides the import for pansql inside the only function that uses it

Both of the hides are for seemingly specific functionality, so I don't think this will bother anyone.

### CODE REVIEW

File: src/sssom/cliques.py [hunk 1]
```diff
@@ -27,9 +27,13 @@
 from .sssom_document import MappingSetDocument
 from .util import MappingSetDataFrame
 
+if TYPE_CHECKING:
+    import networkx
 
-def to_digraph(msdf: MappingSetDataFrame) -> nx.DiGraph:
+def to_digraph(msdf: MappingSetDataFrame) -> "networkx.DiGraph":
```
Comment (matentzn) [inline src/sssom/cliques.py:33]: We really do not need this method at all, but its a good first step to remove the networkx dependency

**Review by matentzn — APPROVED** [review]

File: src/sssom/cliques.py (hunk 1 above)
Comment (cthoyt) [reply src/sssom/cliques.py:33]: I'm not against having a first-party export function to convert SSSOM mappings into a graph I think this is valuable, but having it mixed up with the cliques.py code is not great

**Comment by matentzn** [comment]
@hrshdhgd do you see anything problematic here?

**Review by hrshdhgd — APPROVED** [review]
Looks good to me.

**Review by matentzn — APPROVED** [review]

**Pull request merged**